src/Components/Pages/TodoPage.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { useTodos } from '../Providers/TodoProvider';
+import { useTheme } from '../Providers/ThemeProvider';
+import '../Styles/TodoPage.css';
+import { CircularProgressbarWithChildren, buildStyles } from 'react-circular-progressbar';
+import 'react-circular-progressbar/dist/styles.css';
+import { VscDebugStart } from 'react-icons/vsc';
+import { MdOutlineRestartAlt } from 'react-icons/md';
+import { AiOutlinePause } from 'react-icons/ai';
+
+const TodoPage = () => {
+    const { id } = useParams();
+    const { todos } = useTodos();
+    const { theme } = useTheme();
+    const [ currentTodo, setCurrentTodo ] = useState({});
+    const [ timer, setTimer ] = useState({
+        minutes: 0,
+        seconds: 0
+    });
+    const [ timerOn, setTimerOn ] = useState(false);
+    const [ colorId, setColorId ] = useState('');
+
+    const { minutes, seconds } = timer;
+
+    useEffect(() => {
+        setCurrentTodo({...todos.todos.filter(todo => todo.id === id )[0]});
```

Comment:
you can destructure todos above when you are destructuring it in line 14. So you don't have to do todos.todos.filter. 
Basically you can replace ```const { todos } = useTodos();``` to ```const { todos:{todos} } = useTodos();```